zhaquirks/tuya/thermostat_88teujp.py

```diff
@@ -0,0 +1,251 @@
+
+
+def decimal_to_payload(number):
+    """Coverts decimal to command payload.
+
+    e.g. 295 => [4, 0, 0, 1, 39] and 220 => [4, 0, 0, 0, 220]
+    """
+    hex = "{:X}".format(number).rjust(4, "0")
+    chunk1 = int(hex[0:2], 16)
+    chunk2 = int(hex[2:], 16)
+    return [4, 0, 0, chunk1, chunk2]
+
+
+class ManufacturerThermostatCluster(TuyaManufCluster):
+    """Manufacturer thermostat cluster."""
+
+    def handle_cluster_request(self, tsn, command_id, args):
+        """Handle cluster request."""
+        if command_id != TUYA_GET_DATA and command_id != TUYA_SET_DATA_RESPONSE:
+            return
+
+        tuya_cmd = args[0]
+        decimal = payload_to_decimal(tuya_cmd.data)
+
+        if tuya_cmd.command_id == OCCUPIED_HEATING_SETPOINT_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                OCCUPIED_HEATING_SETPOINT_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == LOCAL_TEMP_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                LOCAL_TEMP_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == SYSTEM_MODE_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                SYSTEM_MODE_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == BATTERY_STATE_COMMAND_ID:
+            self.endpoint.device.battery_bus.listener_event(BATTERY_REPORTED, decimal)
+        elif (
+            tuya_cmd.command_id >= 110 and tuya_cmd.command_id <= 129
+        ):  # Ignore historical data and schedules
+            pass
+        elif tuya_cmd.command_id == 362:  # Ignore away mode
+            pass
+        elif tuya_cmd.command_id == 364:  # Ignore auto/manual mode
+            pass
+        else:
+            _LOGGER.debug(
+                "Unknown command received: %d, %d",
+                tuya_cmd.command_id,
+                decimal,
+            )
+
+
+class PowerConfigurationCluster(LocalDataCluster, PowerConfiguration):
+    """Power configuration cluster."""
+
+    cluster_id = PowerConfiguration.cluster_id
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.battery_bus.add_listener(self)
+
+    def battery_reported(self, value):
+        """Handle reported battery state."""
+        _LOGGER.debug("reported battery alert: %d", value)
+        if value == 1:  # alert
+            self._update_attribute(
+                BATTERY_PERCENTAGE_REMAINING_ATTR, 0
+            )  # report 0% battery
+        else:
+            self._update_attribute(
+                BATTERY_PERCENTAGE_REMAINING_ATTR, 200
+            )  # report 100% battery
+
+
+class ThermostatCluster(LocalDataCluster, Thermostat):
+    """Thermostat cluster."""
+
+    cluster_id = Thermostat.cluster_id
+
+    _CONSTANT_ATTRIBUTES = {
+        MIN_HEAT_SETPOINT_ATTR: 500,
```

Comment:
In my PR, I intend to map the tuya attributes to these, since several devices actually send that information. There would still be a default value for the devices that do not.

Reply:
As far as I can see, my device is a bit stupid, or I couldn't find a way to get this info from it

Reply:
That's why I think I would provide default values, which would then be overwritten as soon as a value is sent by the device, if ever.